storage/namespace.go

```diff
@@ -149,15 +151,24 @@ func newDatabaseNamespace(
 	increasingIndex increasingIndex,
 	writeCommitLogFn writeCommitLogFn,
 	opts Options,
-) databaseNamespace {
-	id := metadata.ID()
+) (databaseNamespace, error) {
+	if err := metadata.Validate(); err != nil {
```

Comment:
I'm not too sure about validating the metadata everywhere. It's an unexported ctor that's only called from database so I'd assume the error checking is done in the database ctor and the metadata passed in is always valid. Handling metadata validation in both places seems kinda redundant.